src/VisualStudio/CSharp/Impl/Options/AdvancedOptionPageControl.xaml.cs

```diff
@@ -130,7 +130,7 @@ public AdvancedOptionPageControl(OptionStore optionStore, IComponentModel compon
             BindToOption(ShowHintsForLambdaParameterTypes, InlineHintsOptions.ForLambdaParameterTypes, LanguageNames.CSharp);
             BindToOption(ShowHintsForImplicitObjectCreation, InlineHintsOptions.ForImplicitObjectCreation, LanguageNames.CSharp);
 
-            BindToOption(ShowInheritanceMargin, FeatureOnOffOptions.ShowInheritanceMargin, LanguageNames.CSharp);
+            ShowInheritanceMargin.IsChecked = OptionStore.GetOption(FeatureOnOffOptions.ShowInheritanceMargin, LanguageNames.CSharp);
```

Comment:
Why a tristate? We've done similar things with https://github.com/dotnet/roslyn/pull/54371/files#diff-59afa2ded2b7a25260638a68dedf3af86b5c731c41fa0f40d645c0c43a092a35R94 if I understand your implementation correctly

Reply:
Your link is pointing to this PR, is that a mistake?

I think this is how we handle the usual experment, is that correct?
We have an tri state checkbox and the default value is null, then check if the experiment is on.
Whenever the user click the checkbox, we change it to two value checkbox, and always respect their choice.

Reply:
it was pointing to earlier in the file because I was lazy and didn't want to find the file in main :) 

```csharp
            BindToOption(Enable_all_features_in_opened_files_from_source_generators, SourceGeneratedFileManager.EnableOpeningInWorkspace, () =>
            {
                // If the option has not been set by the user, check if the option is enabled from experimentation.
                // If so, default to that. Otherwise default to disabled
                return experimentationService?.IsExperimentEnabled(WellKnownExperimentNames.SourceGeneratorsEnableOpeningInWorkspace) ?? false;
            });
```

Reply:
I should have phrased my question/suggestion better: tristate is correct I believe, and we have a BindToOption for it. I think you should use that

Reply:
:) got you

Reply:
OK I change this part. Mark as resolved.